edx-olive/edx-platform
# CAM2-136 Updates translations from Transifex

Updates the translations in the current Olive release branch using the spreadsheet provided by the client.

See notes below on why this had to be manual, and could not be pulled from Transifex.

**JIRA tickets**: CAM2-136 / SE-697

**Screenshots**: 

Signin:
![signin](https://user-images.githubusercontent.com/7556571/53298302-c80def00-387b-11e9-8318-78899f0bddc9.png)

Forgot password:
![forgot password](https://user-images.githubusercontent.com/7556571/53298306-cd6b3980-387b-11e9-8f44-e82a9924d94b.png)

Register:
![register](https://user-images.githubusercontent.com/7556571/53465248-fe768480-3a9c-11e9-9b8f-584cc648fe07.png)

Activation email:
![activation email](https://user-images.githubusercontent.com/7556571/54004850-163dbf00-41a7-11e9-9f14-43e4002deb34.png)

Course home page:
![course home page](https://user-images.githubusercontent.com/7556571/54004624-2608d380-41a6-11e9-9c95-18c7dfcc2d39.png)

Course page:
![course page](https://user-images.githubusercontent.com/7556571/54004711-86981080-41a6-11e9-8ccd-0c1215f52a08.png)

Instructor > Student Admin:
![delete learner s state](https://user-images.githubusercontent.com/7556571/54005152-3de15700-41a8-11e9-9a8a-e6f98e7a1f9e.png)

Instructor Dashboard > Data Downloads:
![data download](https://user-images.githubusercontent.com/7556571/53299434-dcf17f00-3889-11e9-982d-8b9771bd159f.png)

Instructor Dashboard > Discussions:
![discussions](https://user-images.githubusercontent.com/7556571/53299432-d400ad80-3889-11e9-8ac0-3aca8c295609.png)

**Testing instructions**:

Check that all the translated strings in the spreadsheet (see ticket) are present and correct on these screens.

1. To get this branch working with the Ginkgo campus devstack:
   * Snapshot before switching to this branch
   * Remove the custom authentication backends from `lms.env.json`: `"THIRD_PARTY_AUTH_BACKENDS": []` (the Ginkgo version is incompatible with the production Olive branch).
   * Ensure that you're using the Campus production theme branch, [edx-olive:themes/olive/release](github.com:edx-olive/themes/tree/olive/release), and that the lms.env.json is configured to use the theme locale files: 
   ```
   COMPREHENSIVE_THEME_LOCALE_PATHS": [
        "/edx/app/edxapp/themes/conf/locale"
    ],
   ```
1. Run this branch inside a campus devstack, run migrations, and `paver run_all_servers`
1. Visit the home page as an anonymous user, and select Hebrew as your active language
1. Visit the /login page, and Forgot password screens, and ensure there's no English text visible, and check the translations.
1. Visit the /register page, check the translations, and register a new account.
1. Check the registration email text in the console to ensure it is translated as required.
1. Login as a staff user, and visit Instructor Dashboard > Data Download.
   Check the two modified translations there.
1. Visit Instructor Dashboard > Cohorts, and enable cohorts.
1. Visit Instructor Dashboard > Discussions.
    Check the modified translations.

To ensure we're not introducing the dreaded JS translation issue from CAM2-53:
1. Check that there are no differences between `lms/static/js/i18n/he/djangojs.js` and `cms/static/js/i18n/he/djangojs.js`

**Author notes and concerns**:

1. Because of the substantial code differences between the `olive` branch and `master`, it is now impossible to simply pull translations from Transifex master, so changes must be made manually.
   However, we still want the Olive maintainers to update Transifex master, so that their changes persist into release branches.
1. This PR covers only the translation changes in edx-platform, not the ORA2 translation changes.

**Reviewers**
